src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/TdsParserStateObject.cs

```diff
@@ -0,0 +1,1036 @@
+#endif
+                    }
+                    else
+                    {
+                        Debug.Assert(_inBytesUsed == _inBytesRead, "Did not use all data while reading partial header");
+
+                        // Require more data
+                        if (_parser.State == TdsParserState.Broken || _parser.State == TdsParserState.Closed)
+                        {
+                            // NOTE: ReadNetworkPacket does nothing if the parser state is closed or broken
+                            // to avoid infinite loop, we raise an exception
+                            ThrowExceptionAndWarning();
+                            return true;
+                        }
+
+                        if (!TryReadNetworkPacket())
+                        {
+                            return false;
+                        }
+
+                        if (IsTimeoutStateExpired)
+                        {
+                            ThrowExceptionAndWarning();
+                            return true;
+                        }
+                    }
+                } while (_partialHeaderBytesRead != 0); // This is reset to 0 once we have read everything that we need
+
+                AssertValidState();
+            }
+            else
+            {
+                // normal header processing...
+                _messageStatus = _inBuff[_inBytesUsed + 1];
+                _inBytesPacket = (_inBuff[_inBytesUsed + TdsEnums.HEADER_LEN_FIELD_OFFSET] << 8 |
+                                              _inBuff[_inBytesUsed + TdsEnums.HEADER_LEN_FIELD_OFFSET + 1]) - _inputHeaderLen;
+                _spid = _inBuff[_inBytesUsed + TdsEnums.SPID_OFFSET] << 8 |
+                                              _inBuff[_inBytesUsed + TdsEnums.SPID_OFFSET + 1];
+#if !NETFRAMEWORK
+                SqlClientEventSource.Log.TryAdvancedTraceEvent("TdsParserStateObject.TryProcessHeader | ADV | State Object Id {0}, Client Connection Id {1}, Server process Id (SPID) {2}", _objectID, _parser?.Connection?.ClientConnectionId, _spid);
+#endif
+                _inBytesUsed += _inputHeaderLen;
+
+                AssertValidState();
+            }
+
+            if (_inBytesPacket < 0)
+            {
+#if NETFRAMEWORK
+                throw SQL.ParsingError(ParsingErrorState.CorruptedTdsStream);
+#else
+                // either TDS stream is corrupted or there is multithreaded misuse of connection
+                throw SQL.ParsingError();
+#endif
+            }
+
+            return true;
+        }
+
+        // This ensure that there is data available to be read in the buffer and that the header has been processed
+        // NOTE: This method (and all it calls) should be retryable without replaying a snapshot
+        internal bool TryPrepareBuffer()
+        {
+#if NETFRAMEWORK
```

Comment:
There doesn't seem to be any mention of ReliabilitySection in netcore.

Reply:
Nope. Probably because netcore can't be hosted in sql server because it doesn't contain he reliability infrastructure that neftx does. However what I don't know is whether the netfx version of this library will ever be required to be loaded into sql server, it feels unlikely to me but if it ever was then this stuff would be needed.

Could the reliability stuff be stubbed out as no-ops in netcore or is it in the framework not this library?

Reply:
I think the reliability stuff should be kept. At least as a no-op, where possible. I'm pretty sure they don't exist in the netcore code because the functions simply weren't supported on netcore when sqlclient was first ported to it. Given the status of .NET Framework and System.Data.SqlClient, I think it might be likely that SQL does eventually adopt .NET and MDS in the database. I'd hate to have to re-create all the reliability code. As it is, it would need a thorough going-through...

Reply:
With the reliability requirements removed from the netcore codebase I'm pretty sure it's not possible to have netcore run in-process in sql server and never will be. 
However I do agree that the reliability code should be kept. It may be possible to continue to use netfx based versions.